keras_cv/models/classification/classifier.py

```diff
@@ -0,0 +1,110 @@
+# Copyright 2023 The KerasCV Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Image classifier model that works for most use cases."""
```

Comment:
Known variations of the pooling+dense pattern:

ConvNeXT has a LayerNorm layer in between pooling + dense layers
EfficientNet (all 3 variants) has an optional Dropout layer in between pooling + dense layers
MobileNet has a convolutional top with a few layers
VGG's (both 16 and 19) top includes multiple Dense layers
ViT has a Lambda layer instead of a pooling layer.

In general I think this can be solved by subclassing this class, just wanted to document the options.

It would be very nice, though, if Classifier.from_preset("convnext...") could return a ConvNext classifier instead of having to call ConvNextClassifier.from_preset. Wdyt?

Reply:
So maybe `classifier` can just take a `Sequential` of layers optionally?  🤔

Reply:
I originally thought of a design where the task-specific layers where `ImageClassifier` owned a method like `apply_block` that could be overriden by the user/backbone. However, it seemed closer to a classifier factory than a classifier model. All of our models in KerasNLP (whether backbones or tasks) add specific modeling layers.

My second idea was to allow the `preset` property to determine which backbone are compatible, but this wouldn't stop anyone from passing a mobilenet backbone in the generic constructor.

Are you both comfortable with a Classifier factory instead of a specific model?